share/eds/store.go

```diff
@@ -172,6 +180,26 @@ func (s *Store) gc(ctx context.Context) {
 	}
 }
 
+func (s *Store) watchForFailures(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case res := <-s.shardFailures:
+			log.Errorw("removing shard after failure", "key", res.Key, "err", res.Error)
```

Comment:
is there anywhere we log could capture metrics around this, something as basic as a "shardFailureCount"? I am imagining a scenario where a node runner complains that the something is bad and it'd be possible to say "ah, whats the shardFailureCount" or even to graph that somewhere and see when something like a disk was failing as the cause.

i see some metrics stuff in node, but not sure if any are appropriate? this would be a "nice to have"

Reply:
Hmmm, this is supposed to be once in a blue moon event, but if we add a metric people could add alerts for such an event. This would definitely be a good thing, because this removal process is only a bandaid that requires manual attention anyways. @walldiss do you have anything against us adding a metric here?

Reply:
will add